gfx/src/base/device.cpp

```diff
@@ -176,33 +182,48 @@ GFX_API Result Device::submit(uint32_t submitCount, const SubmitInfo* pSubmits)
       }
 
       const CommandBuffer_t::State& state = commandBuffer->state;
-      for (uint32_t drawIdx = 0; drawIdx < state.numDraws; ++drawIdx) {
+      for (size_t drawIdx = 0; drawIdx < state.numDraws; ++drawIdx) {
         Brush brush = state.drawData[drawIdx].brush;
         Vector2D offset = state.drawData[drawIdx].origin;
-        for (uint32_t primIdx = 0; primIdx < state.drawData[drawIdx].numPrims; ++primIdx) {
+        for (size_t primIdx = 0; primIdx < state.drawData[drawIdx].numPrims; ++primIdx) {
           offset += state.drawData[drawIdx].prims[primIdx].offset;
           Glyph glyph = state.drawData[drawIdx].prims[primIdx].glyph;
-
           uint8_t* pData = static_cast<uint8_t*>(glyph->createInfo.buffer->memory->data);
-          for (uint32_t x = 0; x < glyph->createInfo.extent.x; ++x) {
-            for (uint32_t y = 0; y < glyph->createInfo.extent.y; ++y) {
-              if (x + offset.x >= width || y + offset.y >= height) {
-                continue;
+          for (int32_t x = 0; x < glyph->createInfo.extent.x; ++x) {
+            for (int32_t y = 0; y < glyph->createInfo.extent.y; ++y) {
+              Vector2D screenPos = Vector2D{.x = x + offset.x, .y = y + offset.y};
+              switch (renderTarget->createInfo.edgeBehaviour) {
+                case EdgeBehaviour::eClamp: {
+                  if (screenPos.x >= width || screenPos.y >= height || screenPos.x < 0 ||
+                      screenPos.y < 0) {
+                    continue;
+                  }
+                  break;
+                }
+                case EdgeBehaviour::eWrap: {
+                  if (screenPos.x >= width || screenPos.y >= height || screenPos.x < 0 ||
+                      screenPos.y < 0) {
+                    if (screenPos.x >= width || screenPos.x < 0) {
+                      screenPos.x %= width;
```

Comment:
can you get away with just a -=?

Reply:
not if the x position is more than double the width

Reply:
when will this happen? When will we want to draw a shape that's both (a) larger than the width and (b) offset more than 100% of width?